processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -32,13 +32,13 @@
  * provided to the Aggregator through the MetricSelector object, so whatever creates that object gets to choose how
  * the data is actually stored and accessed.
  */
-public interface AggregatorFactory
+public abstract class AggregatorFactory
```

Comment:
to be precise, are you saying that create BaseAggregatorFactory and have it implemented by all but things like HistogramAggregatorFactory (which do not use default implementation of getMergingFactory)?
In that case, whenever we add more methods to interface (even with default implementation) then HistogramAggregatorFactory would need to extend BaseAggregatorFactory .

I am not sure if combination of interface `Blah` and abstract class `BaseBlah` is superior to just abstract `Blah` . However, I will do that if you very strongly believe so. Do you?